srcs/Server.cpp

```diff
@@ -10,30 +10,93 @@
 /* ************************************************************************** */
 
 Server::Server() {}
-Server::Server(const std::string& server_block, const std::string& location_blocks)
-: 
+
+Server::Server(ServerManager* server_manager, const std::string& server_block, std::vector<std::string>& location_blocks, Config* config)
 {
-	this->m_server_name.clear();
-	this->m_host.clear();
-	this->m_port = 0;
-	this->m_fd = 0;
-	this->m_request_uri_limit_size = 0;
-	this->m_request_header_limit_size = 0;
-	this->m_limit_client_body_size = 0;
-	this->m_default_error_page.clear();
+	std::map<std::string, std::string> server_map = ft::stringVectorToMap(ft::split(server_block, '\n'), ' ');
+	struct sockaddr_in server_addr;
+	std::string uri;
+	std::string block;
+	std::string error_page;
+	std::string line;
+
+	m_manager = server_manager;
+	m_config = config;
+	if (ft::hasKey(server_map, "server_name"))
+		m_server_name = server_map["server_name"];
+	else
+		m_server_name = "noname";
+	m_host = server_map["host"];
+	m_port = std::stoi(server_map["port"]);
+	m_request_uri_limit_size = std::stoi(server_map["REQUEST_URI_LIMIT_SIZE"]);
+	m_request_header_limit_size = std::stoi(server_map["REQUEST_HEADER_LIMIT_SIZE"]);
+	m_limit_client_body_size = std::stoi(server_map["LIMIT_CLIENT_BODY_SIZE"]);
+	error_page = server_map["DEFAULT_ERROR_PAGE"];
+	//socket 생성
+	if((m_fd = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP)) == -1)
+		throw "SOCKET ERROR";
+	int value = true;
+	if (setsockopt(m_fd, SOL_SOCKET, SO_REUSEADDR, &value, sizeof(int)) == -1)
+		throw "SOCKET_OPTION ERROR";
+	bzero(&server_addr, sizeof(struct sockaddr_in));
+	// memset((void *)&server_addr, 0x00, sizeof(server_addr));
+	server_addr.sin_family = AF_INET;
+	server_addr.sin_addr.s_addr = inet_addr(m_host.c_str());
+	server_addr.sin_port = htons(8080);
+	if(bind(m_fd, reinterpret_cast<struct sockaddr *>(&server_addr), sizeof(struct sockaddr)) == -1)
+		throw "BIND ERROR";
+	if(listen(m_fd, 10) == -1)
```

Comment:
```suggestion
	if(listen(m_fd, 512/N) == -1)
```

처음에 논의하였을 때 전체 서버의 개수를 세고 난 뒤 512/N으로 listen 하기로 했었는데...생각해보니까 이걸 생성자에서 해버리면 그렇게 못할 것 같네요?! 이슈를 해결해야 할 것 같습니다. 제 생각에는 server.init 과 같은 함수를 따로 제작해서 bind-listen 등은 거기서 하는 것도 방법일 것 같아요

Reply:
64로 수정했습니다